### ISSUE TITLE
document not found

### ISSUE THREAD

**Issue by GodBleak** [couchbaselabs/node-ottoman#634]
Hi, I'm having some difficulties getting started. 
I've pulled the latest docker image for CouchBase, 
ran it, loaded travel-sample, and used the following to try to retrieve a document.
```ts
import {Schema, Ottoman} from "ottoman"

const HotelSchema = new Schema({
    address: { type: String, required: true },
    alias: String,
    checkin: String,
    checkout: String,
    city: { type: String, required: true },
    country: { type: String, required: true },
    description: String,
    directions: [String],
    email: String,
    fax: String,
    free_breakfast: Boolean,
    free_internet: Boolean,
    free_parking: Boolean,
    name: { type: String, required: true },
    pets_ok: Boolean,
    phone: { type: String, validator: 'phone' },
    price: Number,
    public_likes: [String],
    state: String,
    title: String,
    tollfree: String,
    vacancy: Boolean,
})
HotelSchema.index.findByName = { by: 'name', type: 'n1ql' };
let ottoman = new Ottoman()
await ottoman.connect({
    connectionString: 'couchbase://localhost',
    bucketName: 'travel-sample',
    username: 'Administrator',
    password: '7q89X2UUYcwy',
})
const Hotel = ottoman.model('hotel', HotelSchema);
await ottoman.start();

Hotel.findById('hotel_10025')
.then(console.log)
```
However, this produces the error `document not found`.

I've tried to use `.find()` to find it, to find all documents, tried specifying the searchConsistency in the findOptions, and all of which have returned 0 rows. Clearly, I'm missing something, but I can't figure out what. 
What more is required to get a document?

I'm using ottoman 2.1.0
and couchbase 7.0.3

**Comment by gsi-alejandro** [comment]
hi @GodBleak 

please update these lines:
```js
const ottoman = new Ottoman({ scopeName: 'inventory', modelKey: 'type' }); // to use the inventory scope where hotel collection is placed
````

- set `scopeName` to "inventory" to use the inventory scope where hotel collection is placed in travel-sample bucket.
- set `modelKey` to "type", due to the Ottoman's default modelKey value is `_type`.

```js
const Hotel = ottoman.model('hotel', HotelSchema, {keyGeneratorDelimiter: '_'}); 
```

- set keyGeneratorDelimiter to "_", due to Ottoman's default keyGeneratorDelimiter value is ::

then you should be able to retrieve documents:

Get hotel by ID
`const hotel = await Hotel.findById('hotel_10025')`
or in the simplified way
`const hotel = await  Hotel.findById('10025')`

List of hotels
`const hotels = await Hotel.find({}, {limit: 10});`


Notice: These configurations aren't needed to work with Ottoman when you create your own bucket from scratch.

I hope it's helpful, have a nice day.

**Comment by GodBleak** [comment]
Thank you, that works! The issue boiled down to misunderstanding the purpose of `modelKey`.

Is storing a model key a must? My models are based on the collection, so it seems redundant to have my model reflect the collection and need to be tied to every document in the collection with a key.

**Comment by gsi-alejandro** [comment]
Yeah, we're aware, but it remains this way for backward compatibility. We have a plan to clean it soon. (maybe for version 3)

thanks for your feedback, it helps us to improve Ottoman 🎉 .
